### PR TITLE
Make the lab camera better

### PR DESCRIPTION
**Pull request by The-E**
This changes the model rotation method from a "trackball" to a "turntable" model. 
By default, rotations are entirely disabled in the lab. By pressing "R", you can cycle through the rotation modes (Disabled -> Yaw only -> Pitch only -> Yaw and Pitch -> Disabled), the "S" key allows you to set the rotation speed (Fast -> Slow -> Slowest -> Fast).

NOTE: This only affects the model (i.e. rotations while holding down the left mouse button), not the background/camera (i.e. rotations while holding the right mouse button).

### CODE REVIEW

**Comment by asarium** [comment]
With these changes, I cannot rotate the ship with the left mouse button any more. Is this intentional?

**Comment by The-E** [comment]
Sort of?
The rotation control starts off in the "Disabled" state, you need to press R to cycle into a mode where you can actually spin things.

**Comment by asarium** [comment]
I think it's counter-intuitive to have the action of the left mouse button disabled by default. Maybe the camera rotation could be controlled with the left mouse button and the ship rotation with the right?

File: code/lab/lab.cpp [hunk 1]
```diff
@@ -97,77 +94,87 @@ static char Lab_weaponmodel_filename[MAX_SHIP_WEAPONS][MAX_FILENAME_LEN];
...
 static int anim_timer_start = 0;
 
-static float Lab_thrust_len = 1.0f;
+static float Lab_thrust_len      = 1.0f;
 static bool Lab_thrust_afterburn = false;
 
+static int Lab_rotation_mode                   = 0;
+static float Lab_manual_rotation_speed_divisor = 100.f;
+
+#define LAB_ROTATION_MODE_YAW 0
```
Comment (asarium) [inline code/lab/lab.cpp]: These defines should be an `enum class` or a simple `enum` if necessary.

File: code/lab/lab.cpp [hunk 2]
```diff
@@ -97,77 +94,87 @@ static char Lab_weaponmodel_filename[MAX_SHIP_WEAPONS][MAX_FILENAME_LEN];
...
 SCP_string Lab_team_color = "<none>";
 
 camid Lab_cam;
 float lab_cam_distance = 100.0f;
-float lab_cam_phi = 1.24f; // Values chosen to approximate the initial rotation used previously
-float lab_cam_theta = 2.25f;
-bool Lab_render_wireframe = false;
-bool Lab_render_without_light = false;
+float lab_cam_phi      = 0;
+// 1.24f; // Values chosen to approximate the initial rotation used previously
```
Comment (asarium) [inline code/lab/lab.cpp]: These comments are outdated and not necessary any more.

**Comment by asarium** [comment]
@The-E Are there any changes left that you would like to add here?

**Comment by The-E** [comment]
Not at this time.

**Pull request merged**